processor/xprocessor/go.mod

```diff
@@ -0,0 +1,63 @@
+module go.opentelemetry.io/collector/processor/xprocessor
+
+go 1.22.0
+
+require (
+	github.com/stretchr/testify v1.10.0
+	go.opentelemetry.io/collector/component v0.115.0
+	go.opentelemetry.io/collector/consumer/consumertest v0.115.0
+	go.opentelemetry.io/collector/consumer/xconsumer v0.115.0
```

Comment:
please use the commit version of xconsumer after https://github.com/open-telemetry/opentelemetry-collector/pull/11884 is merged

Reply:
Can you approve this PR and I'll do the follow up?

Reply:
If my understanding is correct, we need to do it in this order:
1. merge https://github.com/open-telemetry/opentelemetry-collector/pull/11884
2. get a commit version of `xconsumer ` from mainline and use that version here
3. merge this PR
4. get a commit version of `xprocessor ` from mainline
5. send another PR to update `xprocessor` version

this is because xprocessor depends on xconsumer and right now xconsumer@v0.115.0 does not exist

Reply:
> this is because xprocessor depends on xconsumer and right now xconsumer@v0.115.0 does not exist

Dependencies in core itself doesn't matter because we use replaces here. It's only blocking core->contrib update

Reply:
chatted offline, we'll do the version updates for all new packages in one PR